injector_helper.c

```diff
@@ -88,11 +88,10 @@ static void handle_injector_response(struct evhttp_request *res, void *ctx_void)
 static void forward_request_to_injector(request_ctx *ctx, injector *i);
 
 // Returns true on error
-static bool fd_local_info(int fd, const char **addr, uint16_t *port)
+static bool fd_local_info(int fd, char *addr, size_t addrlen, uint16_t *port)
```

Comment:
> What does this function do?

It just "returns" the local address and port of the socket represented by that `fd`. The code is pretty much a copy/paste from [here](https://github.com/nmathewson/Libevent/blob/2773a5ed5e75540d43500e054581ea1fab2132ac/sample/http-server.c#L377).

> Where does it change addr?

[Here](https://github.com/clostra/dcdn/pull/30/files/59b580a4c034eec858d6b8dd4b0cf529c4a0894d#diff-e90daa1466feff8e719bfc57e56fcc95R116).
> What does addr represent?

The local address of the socket represented by the `fd`. Most likely "0.0.0.0". This string is only used when printing the line `listening on UDP:0.0.0.0:5678`. Getting the local port has more significance though.